### DISCUSSION TITLE
Could this library benefit from System.Buffers and its cohorts?

### DISCUSSION THREAD

**Discussion by JesseTG** [dre0dru/LocalStorage#6] [Ideas]
.NET provides several APIs that can make data management (serialization, storage, etc.) faster and leaner. For instance:

- [System.Buffers](https://docs.microsoft.com/en-us/dotnet/api/system.buffers?view=netstandard-2.1)
- [`System.Memory<T>`](https://docs.microsoft.com/en-us/dotnet/api/system.memory-1?view=netstandard-2.1)
- [`System.Span<T>`](https://docs.microsoft.com/en-us/dotnet/api/system.span-1?view=netstandard-2.1)

These APIs are not provided by Unity, **but** they are all available in a [package registry](https://github.com/xoofx/UnityNuGet) and on [OpenUPM](https://openupm.com/nuget/).

What do you think?

**Comment by dre0dru** [comment]
Can't find `Span<T>` in [regisrty](https://github.com/xoofx/UnityNuGet/blob/master/registry.json), but will check it out nevertheless.
Probably it will become an optional dependency for package if it will work out.

**Reply by JesseTG** [reply]
It's part of the `System.Memory` assembly.
